config/routes.rb

```diff
@@ -43,5 +43,10 @@
 
   get "/redirect/statistics/announcements" => "redirection#upcoming_statistics"
 
+  get "/*slug" => "redirection#covid_topical_event", constraints: lambda { |request|
+    topical_events = request.params["topical_events"]
+    topical_events && topical_events.include?("coronavirus-covid-19-uk-government-response")
+  }, as: :redirect_covid
```

Comment:
If you reorder the elements, I think `do` can be used, which I think makes it a little cleaner:
```ruby
  get "/*slug" => "redirection#covid_topical_event", as: :redirect_covid, constraints: lambda do |request|
    topical_events = request.params["topical_events"]
    topical_events && topical_events.include?("coronavirus-covid-19-uk-government-response")
  end
```
Also if the action is called `redirect_covid` I don't think you need the `:as` declaration

Reply:
using the `lambda do |foo|; end` syntax weirdly gave me 93 errors like this:
```
An error occurred while loading ./spec/validators/param_validator_spec.rb.
Failure/Error: require File.expand_path("../config/environment", __dir__)

FrozenError:
  can't modify frozen Array
# ./config/environment.rb:5:in `<top (required)>'
# ./spec/spec_helper.rb:9:in `<top (required)>'
# ./spec/validators/param_validator_spec.rb:1:in `<top (required)>'
No examples found.
```
They stopped when I wrapped the lambda block in brackets - so I figure the routes parser struggles a bit with it.
I'll stick with the curly braces instead :-)

Reply:
I've taken your other suggestions though, thanks!

Reply:
OK - worth a try. I'm glad my other suggestions were helpful.